draft-oid4vc-haip-sd-jwt-vc.md

```diff
@@ -139,7 +139,42 @@ Note: Issuers should be mindful of how long the usage of the refresh token is al
 
 ### Wallet Attestation Schema {#wallet-attestation-schema}
 
-[Section 3.1 of wallet attestation draft would define the basics, and this profile will define the details.]
+Wallets MUST use attestations following the definition given in [!I-D.ietf-looker-key-attestation-client-authentication].
+
+In addition, the Wallet Attestation MUST contain the following claims:
+
+* `key_type`: this claim asserts the security mechanism the wallet can use to manage private keys. This capability is based on the capabilities of the execution environent of the wallet, this might be a secure element (in case of a wallet residing on a smartphone) or a Cloud-HSM (in case of a cloud wallet). This specification defines the following values for `key_type`: `STRONGBOX`, `TEE`, `SecureEnclave`, `Software`.
+* `user_authentication`: this claim asserts the security mechanism the wallet can use to authenticate access to private keys. This specification defines the following values for `user_authentication`: `APP_PIN_6_DIGITS`, ...
+
+These additional claims inform the issuer about the security capabilities of the wallet and allows the issuer to refuse credential issuance if the achievble security level of a certain wallet does not fulfil the issuer's requirements.
+
+Wallet attestions MUST support web-based key resolution as defined in Section 5 of [@!I-D.terbu-sd-jwt-vc]. The JOSE header `kid` MUST be used to identify the respective key.
+
+This is an example of a wallet attestation:
+
+```json
+{
+  "typ": "wallet-attestation+jwt",
+  "alg": "ES256",
+  "kid": "1"
+}
+.
+{
+  "iss": "https://wallet.example.com",
+  "sub": "https://wallet.example.com",
```

Comment:
I'm not in favor of duplicating iss to sub

I have the strong assumption that the wallet provider and the wallet instance are different entities

In my current implementation we have decided to use the jwk thumbprint to identify the wallet instance and then we use this value in the sub

there's a good agreement to simply remove the sub claim, WDYT about this?

Reply:
> sub descibes the specific Wallet Solution of a Wallet Instance
which includes the option to leave instances anonymous, right?

Reply:
yes, anonymous instances should be RECOMMENDED in my opinion

Reply:
do we assume that a single wallet provider may own multiple wallet solution?

I assume that a single wallet provider owns a single wallet solution.
If we don't have agreement on how the sub value should be valued, considering that it may have the same value of iss and that's misleading since provider and instance are different entities, I suggest to get sub out by simply removing it from the required attributes

Reply:
Is the **Wallet Provider** always the **Attester** of the **Wallet Instance**?

If that is not the case, it means there are three entities, the Wallet Provider, the Wallet Instance and the Attester. It is theoretically impossible to map the three entities to the two claims, `iss` and `sub`, without conflicts.

If the Wallet Provider is NOT always the Attester, the definition of the `iss` claim by eIDAS is problematic and conflicts with the specification of the attestation-based client authentication.

Reply:
According to the underlying draft-looker-oauth-attestation-based-client-auth, the `iss` is the entity the AS (here the wallet) trusts for the purpose of the issuance of attestations and the `sub` is the client id. The wallet can establish the trust be looking the `iss` value up in a trusted list, for example. The actual wallet product could be the `sub` value. The attributes (including capabilities and perhaps even authorization) need to be conveyed through claims in the attestation. 
The difference between attester and wallet provider would be that if the wallet provider is in the iss, it needs to be entitled by the trusted list to issue it's own attestations whereas in the case of the attester, it's the attester who has that entitlement. 

Does this sound reasonable?

Reply:
According to the underlying draft-looker-oauth-attestation-based-client-auth, the `iss` is the entity the AS (here the wallet) trusts for the purpose of the issuance of attestations and the `sub` is the client id. The wallet can establish the trust be looking the `iss` value up in a trusted list, for example. The actual wallet product could be the `sub` value. The attributes (including capabilities and perhaps even authorization) need to be conveyed through claims in the attestation. 
The difference between attester and wallet provider would be that if the wallet provider is in the iss, it needs to be entitled by the trusted list to issue it's own attestations whereas in the case of the attester, it's the attester who has that entitlement. 

Does this sound reasonable?

Reply:
I'm in favor of having sub as optional and without any constraint about its value

the current proposal is concerning to me

Reply:
in the call yesterday we came to the conclusion that the fundamental question is:
"Is the Wallet Provider always the Attester of the Wallet Instance?"

Indeed, we have already seen the reality that wallet providers may not have the capacity/willingness to run an attestation service on their own. And nobody will force an wallet provider to use a different attestation service. But we should keep this option open. Therefore I am voting to use:
- `iss` as the issuer of the wallet attestation
- `sub` as the client id
The version should not be referenced in there as this is responsibility of the wallet provider and not a burden to the issuer. If the Wallet Provider runs the attestation service on his own, we have two possibilities:
- a) `iss` equals `sub`
- b) `sub` may be omitted
I'm in favor of a)

Reply:
@peppelinux making the `sub` optional cannot be done in HAIP and needs to be discussed in the IETF on the client attestation draft. For HAIP, I suggest we remove the concrete values and put in letter what the values are expected to be: code suggestion below.

Reply:
in the implementation I'm working on the `sub` value is the thumbprint of the cnf.jwk since this latter uniquely identify, anyway, the wallet instance